diff --git a/lib/commands/build/plugin.js b/lib/commands/build/plugin.js
index 3b4fd06aa167df7533f9ef577242a18207fab223..e4b1a75c7dfc09f2530f9eb5aea8465b8e95a5e9 100644
--- a/lib/commands/build/plugin.js
+++ b/lib/commands/build/plugin.js
@@ -65,6 +65,35 @@ class BuildPluginCommand extends clipanion_1.Command {
                         });
                     },
                 };
+                const runtimeLoader = {
+                    name: 'runtime-loader',
+                    setup(build) {
+                        build.onResolve({ filter: /^@monstrs\/yarn-runtime$/ }, args => ({
+                            path: args.path,
+                            namespace: 'yarn-runtime'
+                        }))
+
+                        build.onResolve({ filter: /^@monstrs\/yarn-runtime\/dist$/ }, args => ({
+                            path: args.path,
+                            external: true
+                        }))
+
+                        build.onLoad({ filter: /.*/, namespace: 'yarn-runtime' }, async (args) => ({
+                            contents: `const runtime = () => {
+                                try {
+                                    return require('@monstrs/yarn-runtime/dist')
+                                } catch {
+                                    require(process.cwd() + '/.pnp.cjs').setup()
+
+                                    return require('@monstrs/yarn-runtime/dist')
+                                }
+                            }
+
+                            module.exports = runtime()`,
+                            loader: `js`,
+                        }))
+                    }
+                }
                 const res = await (0, esbuild_1.build)({
                     banner: {
                         js: [
@@ -89,7 +118,7 @@ class BuildPluginCommand extends clipanion_1.Command {
                     logLevel: `silent`,
                     format: `iife`,
                     platform: `node`,
-                    plugins: [dynamicLibResolver, (0, esbuild_plugin_pnp_1.pnpPlugin)()],
+                    plugins: [dynamicLibResolver, runtimeLoader, (0, esbuild_plugin_pnp_1.pnpPlugin)()],
                     minify: !this.noMinify,
                     sourcemap: this.sourceMap ? `inline` : false,
                     target: `node12`,